### PR TITLE
basic pixel throwing, space drift and more pixel projectile stuff

### PR DESCRIPTION
**Pull request by Couls**
and basic pixel mechs and pointing. Pretty rough with a lot of commented out code and some unused bits(like the x and y position of the mouse for throwing) that might be able to be used later

### CODE REVIEW

File: code/controllers/subsystem/spacedrift.dm [hunk 1]
```diff
@@ -1,7 +1,7 @@
 SUBSYSTEM_DEF(spacedrift)
 	name = "Space Drift"
 	priority = FIRE_PRIORITY_SPACEDRIFT
-	wait = 5
+	wait = 1
```
Comment (Couls) [inline code/controllers/subsystem/spacedrift.dm:4]: runs more often than it did before, much smoother

File: code/game/objects/effects/effect_system/effects_other.dm [hunk 2]
```diff
@@ -46,16 +44,29 @@
...
-				E.icon_state = ""
-			if(qdel_in_time)
-				QDEL_IN(E, qdel_in_time)
-	oldposition = get_turf(holder)
+	if(cooldown < world.time)
+		if(!holder.has_gravity() || !nograv_required)
+			if(ismovableatom(holder))
+				var/atom/movable/AM = holder
+				for(var/turf/K in AM.locs)
+					effects_list |= new effect_type(K)
```
Comment (Couls) [inline code/game/objects/effects/effect_system/effects_other.dm]: creates a jetpack effect on every turf the jetpacker is on, looks a bit better than an off-center trail
Though I could center it on the jetpacker using stepx stepy

File: code/controllers/subsystem/spacedrift.dm [hunk 3]
```diff
@@ -28,32 +28,24 @@ SUBSYSTEM_DEF(spacedrift)
...
 		if (!AM.inertia_dir)
-			AM.inertia_last_loc = null
 			processing -= AM
 			if (MC_TICK_CHECK)
 				return
 			continue
 
 		var/old_dir = AM.dir
-		var/old_loc = AM.loc
+		// var/old_loc = AM.loc
```
Comment (ninjanomnom) [inline code/controllers/subsystem/spacedrift.dm]: Just delete instead of commenting it out

File: code/controllers/subsystem/spacedrift.dm [hunk 4]
```diff
@@ -28,32 +28,24 @@ SUBSYSTEM_DEF(spacedrift)
...
 		AM.inertia_moving = TRUE
-		step(AM, AM.inertia_dir)
+		step(AM, AM.inertia_dir, AM.step_size) // TODO: rework inertia to use degstep and angles
 		AM.inertia_moving = FALSE
-		AM.inertia_next_move = world.time + AM.inertia_move_delay
-		if (AM.loc == old_loc)
-			AM.inertia_dir = 0
+/* 		if (AM.loc == old_loc)
+			AM.inertia_dir = 0 */
```
Comment (ninjanomnom) [inline code/controllers/subsystem/spacedrift.dm]: Ditto

File: code/controllers/subsystem/throwing.dm [hunk 5]
```diff
@@ -126,7 +132,7 @@ SUBSYSTEM_DEF(throwing)
 /datum/thrownthing/proc/finalize(hit = FALSE, target=null)
 	set waitfor = FALSE
 	//done throwing, either because it hit something or it finished moving
-	stack_trace("[thrownthing] hit something at [get_turf(thrownthing)]")
+	//stack_trace("[thrownthing] hit something at [get_turf(thrownthing)]")
```
Comment (ninjanomnom) [inline code/controllers/subsystem/throwing.dm]: ditto

File: code/game/atoms_movable.dm [hunk 6]
```diff
@@ -423,8 +423,9 @@
 	if(throwing)
 		return 1
 
-	if(!isturf(loc))
-		return 1
+	for(var/x in locs)
```
Comment (ninjanomnom) [inline code/game/atoms_movable.dm]: You should still only need the loc check, this is just making sure it isnt inside an object

File: code/game/objects/effects/effect_system/effects_other.dm (hunk 2 above)
Comment (ninjanomnom) [reply code/game/objects/effects/effect_system/effects_other.dm]: I would rather you have a single trail effect and offset it behind them

File: code/modules/mob/living/carbon/carbon_movement.dm [hunk 7]
```diff
@@ -8,8 +8,10 @@
 /mob/living/carbon/Process_Spacemove(movement_dir = 0)
 	if(..())
 		return 1
-	if(!isturf(loc))
-		return 0
+		
+	for(var/x in locs)
```
Comment (ninjanomnom) [inline code/modules/mob/living/carbon/carbon_movement.dm]: Again this is just making sure the mob isnt inside something

File: code/modules/projectiles/projectile.dm [hunk 8]
```diff
@@ -468,19 +468,18 @@
 	for(var/i in 1 to SSprojectiles.global_iterations_per_move)
 		if(QDELETED(src))
 			return
-		trajectory.increment(trajectory_multiplier)
-		var/turf/T = trajectory.return_turf()
-		if(T.z != loc.z)
+		//trajectory.increment(trajectory_multiplier)
+		//var/turf/T = trajectory.return_turf()
```
Comment (ninjanomnom) [inline code/modules/projectiles/projectile.dm:472]: More comments to remove

File: code/modules/projectiles/projectile.dm [hunk 9]
```diff
@@ -468,19 +468,18 @@
...
 		if(QDELETED(src))
 			return
-		trajectory.increment(trajectory_multiplier)
-		var/turf/T = trajectory.return_turf()
-		if(T.z != loc.z)
+		//trajectory.increment(trajectory_multiplier)
+		//var/turf/T = trajectory.return_turf()
+		degstep(src, original_angle, 1)
+		hitscan_last = loc
+/* 		if(T?.z != loc?.z)
```
Comment (ninjanomnom) [inline code/modules/projectiles/projectile.dm:475]: More

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/projectile.dm (hunk 9 above)
Comment (Couls) [reply code/modules/projectiles/projectile.dm:475]: For this one it kind of removes previous functionality, projectiles won’t be able to cross Z levels

**Pull request merged**